libyul/optimiser/UnusedFunctionArgumentPruner.h

```diff
@@ -0,0 +1,48 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * UnusedFunctionArgumentPruner: Optimiser step that removes unused-parameters in a function.
+ *
+ * If a parameter is unused, like in, `function f(a,b,c) -> x { x := div(a,b) }`
+ *
+ * We remove the parameter and create a new "linking" function as follows:
+ *
+ * `function f(a,b) -> x { x := div(a,b) }`
```

Comment:
By the way, we can do the same with unused return variables, right? I think it would be fine to do it in the same component.